modules/azerionedgeRtdProvider.js

```diff
@@ -106,10 +109,79 @@ export function setAudiencesToBidders(reqBidsConfigObj, config, audiences) {
  * @return {boolean}
  */
 function init(config, userConsent) {
-  attachScript(config);
+  if (hasUserConsented(userConsent)) {
+    attachScript(config);
+  }
   return true;
 }
 
+/**
+ * List the vendors consented coming from userConsent object.
+ *
+ * @param {Object} userConsent
+ *
+ * @return {Array}
+ */
+function getVendorsConsented(userConsent) {
+  const consents = userConsent?.gdpr?.vendorData?.vendor?.consents || {};
+  return Object.entries(consents).reduce((acc, [vendorId, consented]) => {
+    return consented ? [...acc, vendorId] : acc;
+  }, []);
+}
+
+/**
+ * List the purposes consented coming from userConsent object.
+ *
+ * @param {Object} userConsent
+ *
+ * @return {Array}
+ */
+export function getPurposesConsented(userConsent) {
+  const consents = userConsent?.gdpr?.vendorData?.purpose?.consents || {};
+  return Object.entries(consents).reduce((acc, [purposeId, consented]) => {
+    return consented ? [...acc, purposeId] : acc;
+  }, []);
+}
```

Comment:
Isn't this something the `tcfConsentManagement` module should handle?

Reply:
I'm not sure we understood the functionality of the `consentManagement` module correctly. Please let me know if anything that I'm saying is incorrect.

The consentManagment module is used to fetch, decode and pass the user consent to bid adapter, RTD modules, etc. The publisher uses this module to configure the integration with its CMP. It also allows adding some rules to automatically enforce GDPR based on their GVL_ID through the gdprEnforcement module. Like the one shown here as Sirdata's example:

```javascript
pbjs.setConfig({
    consentManagement: {
        usp: {
            cmpApi: 'iab',
            timeout: 100 // US Privacy timeout 100ms
        },
        gdpr: {
          cmpApi: 'iab',
          timeout: 8000,
          defaultGdprScope: true,
          rules: [{
            purpose: "storage",
            enforcePurpose: true,
            enforceVendor: true,
            vendorExceptions: ["appnexus"] //Can work without consent for cookies
          },{
            purpose: "basicAds",
            enforcePurpose: true,
            enforceVendor: true,
            vendorExceptions: ["appnexus"]
          }]
        },
    }
});
```

But if we want to automatically check the consent given for our module from inside our RTD module, without the publisher having to configure that, we can't use the `consentManagement` module. We've seen other modules doing the checks manually like in our case. For instance [sirdata](https://github.com/prebid/Prebid.js/blob/69d79062781c38d8fafcac189f807e99c4b28f65/modules/sirdataRtdProvider.js#L371).

We are not sure if there's a better way of doing this through any helpers, or exported functions from consentManagement/gdprEnforcement modules. The closest we could find was the method [hasPurpose1Consent](https://github.com/prebid/Prebid.js/blob/69d79062781c38d8fafcac189f807e99c4b28f65/src/utils/gpdr.js#L9) but it only checks for one purpose and we need to check five.

Any insight or suggestion about how to better approach this would be greatly appreciated.